utils/run_test.sh

```diff
@@ -248,7 +248,9 @@ if [ -d "/mnt/daos" ]; then
         for ((i = 0; i < ${#failures[@]}; i++)); do
             echo "    ${failures[$i]}"
         done
-        exit 1
+        if ! ${IS_CI:-false}; then
```

Comment:
The only other place IS_CI use used looks like 'if [ -n "${IS_CI:-}" ]; then', I would expect these two to be the same, although I don't know which is correct.

Reply:
They will both work for >99% of the use cases.  My use makes it `false` if it's empty and does a boolean test with it.  The use above makes it empty if it's empty (which is kind of strange, indeed -- but it's what you have to do with `set -u` and does an emptiness test with it.  The <1% use-case where this will not work above is if somebody/thing sets `IS_CI=false`, which is quite unlikely I think.

It would be ideal if both use-cases worked the same and given that the value is being set as a boolean, the above should probably be changed.  I can put that into my next PR if that's acceptable.  Or I can re-push to this PR if you wish.

Reply:
Nothing is blocked on on this that I'm aware of so I'd rather we got it right first time.

Reply:
This is actually resolving @kprantis's request that timed builds don't stop when there are UT failures as there are apparently enough intermittent failures that they feel that they are missing data from timed runs that stop at UT.

As such this task was given to me from @raydennis as my newest number 1 priority.  I guess we can let either of them decide how urgent this is to land.

Reply:
UT test failures aren't a daily occurrence, re-pushing this today and landing it on Monday isn't going to cause any disruption.

Reply:
Fixed.